src/react-components/room/InviteLinkInputField.js

```diff
@@ -0,0 +1,61 @@
+import React, { useState, useCallback } from "react";
+import PropTypes from "prop-types";
+import styles from "./RoomSettingsSidebar.scss";
+import { IconButton } from "../input/IconButton";
+import { FormattedMessage } from "react-intl";
+import { CopyableTextInputField } from "../input/CopyableTextInputField";
+
+export function InviteLinkInputField({ fetchingInvite, inviteUrl, onRevokeInvite }) {
+  const [showRevokeConfirmation, setShowRevokeConfirmation] = useState(false);
+
+  const revokeInvite = useCallback(() => {
+    setShowRevokeConfirmation(true);
+  }, []);
+
+  const cancelConfirmRevokeInvite = useCallback(() => {
+    setShowRevokeConfirmation(false);
+  }, []);
+
+  const confirmRevokeInvite = useCallback(
+    () => {
+      onRevokeInvite();
+      setShowRevokeConfirmation(false);
+    },
+    [onRevokeInvite]
+  );
+
+  return (
+    <CopyableTextInputField
+      label="Invite link"
+      disabled={fetchingInvite}
+      value={fetchingInvite ? "Generating invite..." : inviteUrl}
```

Comment:
Reminder to add i18n if you haven't gotten to that yet.

Reply:
Yeah, currently putting that on pause so I can unblock some of the QA testing, but I'll come back to it soon.